src/app/api/v1/admin/feature-flags/[flagId]/route.ts

```diff
@@ -0,0 +1,28 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { NextRequest, NextResponse } from "next/server";
+import { getToken } from "next-auth/jwt";
+import { getFeatureFlagByName } from "../../../../../../db/tables/featureFlags";
+import { isAdmin } from "../../../../utils/auth";
+import appConstants from "../../../../../../appConstants";
+
+export async function GET(req: NextRequest) {
+  const token = await getToken({ req });
+  if (typeof token?.email === "string" && isAdmin(token?.email)) {
+    // Signed in
+    const flagName = req.nextUrl.toString().split("/").pop() || "";
```

Comment:
`req.nextUrl.pathname` would be easier/safer I think (I think query params might break this?)

Reply:
Actually looking into this, I think this is the right way to get `flagId` when using the `[]` syntax:
https://nextjs.org/docs/pages/building-your-application/routing/dynamic-routes

Reply:
The `GET` function can take a second parameter, which is an object with the `params` property that contains the `flagId` property from the URL: https://nextjs.org/docs/app/building-your-application/routing/router-handlers#dynamic-route-segments

(Note that `/pages/` in your URL and the "Using Pages Router" in the top left-hand corner of the page you linked; Next.js is transitioning from one routing system to another, and we're already on the new one, the App Router. So make sure to look at the correct version of the docs.)

Reply:
Thanks, I knew there'd be a way to do this! Fixed